### ISSUE TITLE
bed mesh result weird... is this normal?

### ISSUE THREAD

**Issue by tjengbudi** [markniu/Bed_Distance_sensor#139]
after install bd sensor and place mesh bed for 30 point to check
i got strange result. is this normal?
![klipper mesh](https://github.com/markniu/Bed_Distance_sensor/assets/3039807/d4add474-c054-4916-9b26-6543762b3c1a)

**Comment by markniu** [comment]
yes, this looks strange. are the heightmap almost the same for each bed mesh? could you paste some data here from the console?

**Comment by tjengbudi** [comment]
suddenly the bd sensor cannot response. i do not why. now i event cannot get Z HOME... check it first and confirm after 
will report you

**Comment by tjengbudi** [comment]
@markniu i check the cable and change the SDA to used EXP1 port and it still cannot receive any data.
what i must to do?

**Comment by markniu** [comment]
Both the SCL and SDA pins are changed to EXP1? make sure the 2 GND wires are connected to the mainboard

**Comment by tjengbudi** [comment]
the scl still used probe/servos pin.
i checked with m119 the z always open
i put metal and the light in the bdsensor's led is on, but m119 return z
still open

**Comment by markniu** [comment]
M102 S-1 return nothing?

**Comment by tjengbudi** [comment]
it is return "BDsensorVer:V1.1 pandapi3d.com
,switch_mode=0,collision_homing=0,collision_cal=0"

**Comment by tjengbudi** [comment]
M102 S-2 always return " BDsensor:Out of measure Range or too close to the bed"

**Comment by markniu** [comment]
Calibrate again

**Comment by tjengbudi** [comment]
how to calibrate. i cannot homing for Z. it never detect my bed. and crash after i homing. force manual?

**Comment by tjengbudi** [comment]
> Calibrate again

BDsensor is too far from the bed:1015

**Comment by markniu** [comment]
> how to calibrate. i cannot homing for Z. it never detect my bed. and crash after i homing. force manual?

yes ,use the manual and SET_KINEMATIC_POSITION , M102 S-6

**Comment by markniu** [comment]
and then check the data after M102 S-5

**Comment by tjengbudi** [comment]
> > how to calibrate. i cannot homing for Z. it never detect my bed. and crash after i homing. force manual?
> 
> yes ,use the manual and SET_KINEMATIC_POSITION , M102 S-6

after i send M102 S-6 i got BDsensor is too far from the bed:1015

after i send M102 S-5 i got 

"
02:46:26 pm
1015
02:46:26 pm
1015
02:46:26 pm
1015
02:46:26 pm
979
02:46:26 pm
979
02:46:26 pm
979
02:46:26 pm
964
02:46:26 pm
943
02:46:26 pm
923
02:46:26 pm
902
02:46:26 pm
882
02:46:26 pm
860
02:46:26 pm
982
02:46:26 pm
977
02:46:26 pm
950
02:46:26 pm
922
02:46:26 pm
895
02:46:26 pm
907
02:46:26 pm
879
02:46:26 pm
686
02:46:26 pm
663
02:46:26 pm
641
02:46:26 pm
617
02:46:26 pm
594
02:46:26 pm
569
02:46:26 pm
545
02:46:26 pm
519
02:46:26 pm
396
02:46:26 pm
265
02:46:26 pm
587
02:46:26 pm
588
02:46:26 pm
558
02:46:26 pm
528
02:46:26 pm
357
02:46:26 pm
306
02:46:26 pm
280
02:46:26 pm
252
02:46:26 pm
222
02:46:26 pm
195
02:46:26 pm
175
02:46:26 pm
M102 S-5
"

**Comment by markniu** [comment]
the data looks right.  btw the led on represent that it is in the measure range but not mean it is triggered for the z axis

**Comment by tjengbudi** [comment]
and now it suddenly "BDsensor is too far from the bed:1015" again

**Comment by markniu** [comment]
you can comment out the `no_stop_probe` in the section BDsensor, and send M102 S-2 to check the actual distance after it stopped by that error

**Comment by tjengbudi** [comment]
after some touch here and touch there... it is working again.
check cable with multitester. it is connected ok

after plug again and test it is run
but the result is amaizing

![klipper 2](https://github.com/markniu/Bed_Distance_sensor/assets/3039807/cc8c1807-d78f-446d-81f6-3d1dcf7dd8da)

M102 S-2 fave result "0.42mm"

**Comment by markniu** [comment]
is the data returned by M102 S-2 almost the same after very 
G28  
G1 Z1   
M102 S-2

**Comment by tjengbudi** [comment]
after idle... it stop working again... frustrated what happened with this sensor

it crash after i sent g28

**Comment by tjengbudi** [comment]
after hard reboot 



03:48:53 pm
0.42mm
03:48:40 pm
G28
G1 Z1
M102 S-2
03:48:32 pm
0.42mm
03:48:19 pm
G28
G1 Z1
M102 S-2
03:48:10 pm
0.41mm
03:47:58 pm
G28
G1 Z1
M102 S-2


0.41 mm
0.42 mm
0.42 mm trying 3 times

**Comment by tjengbudi** [comment]
trying check and meassure without changing the height at all

the result very strange "
03:53:00 pm
0.44mm
03:53:00 pm
M102 S-2
03:52:56 pm
0.44mm
03:52:56 pm
M102 S-2
03:52:56 pm
1.03mm
03:52:56 pm
M102 S-2
03:52:55 pm
1.03mm
03:52:55 pm
M102 S-2
03:52:55 pm
1.03mm
03:52:55 pm
M102 S-2
03:52:54 pm
1.03mm
03:52:54 pm
M102 S-2
03:52:53 pm
1.03mm
03:52:53 pm
M102 S-2
03:52:52 pm
1.03mm
03:52:52 pm
M102 S-2
03:52:52 pm
1.03mm
03:52:52 pm
M102 S-2
03:52:52 pm
1.03mm
03:52:52 pm
M102 S-2
03:52:51 pm
1.03mm
03:52:51 pm
M102 S-2
03:52:51 pm
1.03mm
03:52:51 pm
M102 S-2
03:52:51 pm
1.03mm
03:52:51 pm
M102 S-2
03:52:51 pm
1.03mm
03:52:51 pm
M102 S-2
03:52:50 pm
1.03mm
03:52:50 pm
M102 S-2
03:52:50 pm
1.03mm
03:52:50 pm
M102 S-2
03:52:50 pm
1.03mm
03:52:50 pm
M102 S-2
03:52:50 pm
1.03mm
03:52:50 pm
M102 S-2
03:52:49 pm
1.03mm
03:52:49 pm
M102 S-2
03:52:49 pm
1.03mm
03:52:49 pm
M102 S-2
03:52:48 pm
1.03mm
03:52:48 pm
M102 S-2
03:52:48 pm
1.03mm
03:52:48 pm
M102 S-2
03:52:47 pm
1.03mm
03:52:47 pm
M102 S-2
03:52:47 pm
1.03mm
03:52:47 pm
M102 S-2
03:52:46 pm
1.03mm
03:52:46 pm
M102 S-2
03:52:45 pm
1.03mm
03:52:45 pm
M102 S-2
03:52:44 pm
1.03mm
03:52:44 pm
M102 S-2
03:52:43 pm
1.03mm
03:52:43 pm
M102 S-2
03:52:42 pm
1.03mm
03:52:42 pm
M102 S-2
03:52:41 pm
0.99mm
03:52:41 pm
M102 S-2
03:52:29 pm
0.99mm
03:52:29 pm
M102 S-2
"

it can changing by him self from 0.99 --> 1.03 -->0.44

**Comment by tjengbudi** [comment]
must i reflash it ?

**Comment by markniu** [comment]
I suspend that there is a hardware problem maybe the cable EMI from other wires. where and when do you purchase this v1.1 . I would like to send you a new one, you can send me your shipping address to my email niujl123@sina.com

**Comment by markniu** [comment]
did the hotend heat? if the hotend too close it maybe heat the coil 

> it can changing by him self from 0.99 --> 1.03 -->0.44

**Comment by tjengbudi** [comment]
No, cold bed...

**Comment by tjengbudi** [comment]
> I suspend that there is a hardware problem maybe the cable EMI from other wires. where and when do you purchase this v1.1 . I would like to send you a new one, you can send me your shipping address to my email [niujl123@sina.com](mailto:niujl123@sina.com)

already sent please take a look
